Aos 2 fixes efif

efifogel opened the pull request:
## Fixed the Landmark point-location strategy of the Aos_2 package, so that it can be applied to arrangements on a sphere.

The title says it all.
I slightly changed one paragraph in the user manual. 
Here is the new version:

The arrangement attached to the landmark strategy must be either (i) an instance of the `Arrangement_2]<Geom,Dcel>` class template, where the `Traits` parameter is substituted by a geometry-traits class that models the `ArrangementLandmarkTraits_2`  concept, or (ii) an instance of the `Arrangement_on_surface_2<GeomTraits,TopolTraits>` class template, where the `GeometryTraits` is similarly substituted; see Section The Landmark Concept for details about this concept.

## Release Management

* Affected package(s): Arrangement_on_surface_2
* Issue(s) solved (if any): NA
* Feature/Small Feature (if any): NA
* Link to compiled documentation: Changed one paragraph; see above.
* License and copyright ownership: TAU

sloriot commented:
```
/home/runner/work/cgal/cgal/Arrangement_on_surface_2/include/CGAL/Arr_point_location/Arr_landmarks_pl_impl.h:28:8: error: redefinition of default argument for ‘class<template-parameter-1-2>’
   28 | struct has_is_on_x_identification_2 : std::false_type {};
```

efifogel commented:
Fixed. (I assume it happens on Windows, or at least, not on my platform,
but it turns out that this construct is not used at all, so I simply
removed it...)
Please test again. Thank you.
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/




On Wed, 18 Dec 2024 at 14:13, Sebastien Loriot ***@***.***>
wrote:

> /home/runner/work/cgal/cgal/Arrangement_on_surface_2/include/CGAL/Arr_point_location/Arr_landmarks_pl_impl.h:28:8: error: redefinition of default argument for ‘class<template-parameter-1-2>’
>    28 | struct has_is_on_x_identification_2 : std::false_type {};
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/CGAL/cgal/pull/8666#issuecomment-2551166502>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABVBNOCFSESVOID25ZNLX3T2GFRH3AVCNFSM6AAAAABTZJYKJGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKNJRGE3DMNJQGI>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

sloriot commented:
@efifogel 
here are platforms failing that could be related to this PR. 

https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-6.1-Ic-59/Arrangement_on_surface_2/TestReport_Christo_MSVC2017-Debug-64bits.gz
https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-6.1-Ic-59/Arrangement_on_surface_2_Examples/TestReport_Christo_MSVC2017-Debug-64bits.gz

Could you please look at the errors and let me know if you think it can be related to your changes?

efifogel commented:
Yes, it's the Aos_2 code.
I pushed a possible fix.
Please test.
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/




On Thu, 9 Jan 2025 at 19:55, Sebastien Loriot ***@***.***>
wrote:

> @efifogel <https://github.com/efifogel>
> here are platforms failing that could be related to this PR.
>
>
> https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-6.1-Ic-59/Arrangement_on_surface_2/TestReport_Christo_MSVC2017-Debug-64bits.gz
>
> https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-6.1-Ic-59/Arrangement_on_surface_2_Examples/TestReport_Christo_MSVC2017-Debug-64bits.gz
>
> Could you please look at the errors and let me know if you think it can be
> related to your changes?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/CGAL/cgal/pull/8666#issuecomment-2580930249>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABVBNODZ7OD6JJDKIE4QVXD2J2Z2LAVCNFSM6AAAAABTZJYKJGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKOBQHEZTAMRUHE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

sloriot commented:
CGAL-6.1-Ic-63 still seem to have the same issues

efifogel commented:
It seems that the issue in the Aos_2 package have been addressed.
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/




On Wed, 15 Jan 2025 at 20:04, Sebastien Loriot ***@***.***>
wrote:

> CGAL-6.1-Ic-63 still seem to have the same issues
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/CGAL/cgal/pull/8666#issuecomment-2593611123>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABVBNOECCBYZOZQW42OWUWT2K2PKFAVCNFSM6AAAAABTZJYKJGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKOJTGYYTCMJSGM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

sloriot commented:
@efifogel  I still see them [here](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-6.1-Ic-63/Arrangement_on_surface_2/TestReport_Christo_MSVC2017-Debug-64bits.gz) and [there](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-6.1-Ic-63/Arrangement_on_surface_2_Examples/TestReport_Christo_MSVC2017-Debug-64bits.gz).

efifogel commented:
ops, sorry. I looked at the wrong file.
Anyhow, I fixed the problem.
It is more of the same: MSVC complains when auto is used for "iterators".
1. It does not complain about all the issues at once, so I fix what I see.
Eventually, I'll get them all...
2. At some point we should understand why MSVC complains, and perhaps fix
the definitions of these problematic types.

   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/




On Thu, 16 Jan 2025 at 08:59, Sebastien Loriot ***@***.***>
wrote:

> @efifogel <https://github.com/efifogel> I still see them here
> <https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-6.1-Ic-63/Arrangement_on_surface_2/TestReport_Christo_MSVC2017-Debug-64bits.gz>
> and there
> <https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-6.1-Ic-63/Arrangement_on_surface_2_Examples/TestReport_Christo_MSVC2017-Debug-64bits.gz>
> .
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/CGAL/cgal/pull/8666#issuecomment-2594674300>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABVBNOG4GSYF64LYLYGP2KD2K5KGXAVCNFSM6AAAAABTZJYKJGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDKOJUGY3TIMZQGA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

sloriot commented:
Successfully tested in CGAL-6.1-Ic-66
